src/Std/Data/TreeSet/Raw/WF.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+prelude
+import Std.Data.TreeMap.Raw.WF
+import Std.Data.TreeSet.AdditionalOperations
+import Std.Data.TreeSet.Raw.Basic
+
+/-!
+# Well-formedness proofs for raw tree sets
+
+This file contains well-formedness proofs about `Std.Data.TreeSet.Raw.Basic`. Most of the lemmas require
+`TransCmp cmp` for the comparison function `cmp`.
+-/
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+universe u v
+
+namespace Std.TreeSet.Raw.WF
+
+open TreeMap.Raw renaming WF → InnerWF
+
+variable {α : Type u} {cmp : α → α → Ordering} {t : Raw α cmp}
+
+theorem empty : (empty : Raw α cmp).WF :=
+  letI : Ord α := ⟨cmp⟩; ⟨InnerWF.empty⟩
```

Comment:
I haven't tested what happens when you remove it, but I'm surprised this local instance is needed. Same for the corresponding lemma in `TreeSet.Raw.WF`.

Reply:
You are correct!